src/compiler/scala/tools/nsc/typechecker/Typers.scala

```diff
@@ -4876,6 +4876,8 @@ trait Typers extends Adaptations with Tags with TypersTracking with PatternTyper
           case LookupInaccessible(sym, msg) => issue(AccessError(tree, sym, context, msg))
           case LookupNotFound               =>
             inEmptyPackage orElse lookupInRoot(name) match {
+              // ignore wildcard bodies inserted by the Java parser
+              case NoSymbol if context.unit.isJava && name == termNames.WILDCARD => tree
```

Comment:
the java parser (`compiler/scala/tools/nsc/javac/JavaParser.scala`) does the following substitutions that result in wildcards:
```
method body -> blankExpr
formalParam -> varDecl -> blankExpr
blankExpr -> Ident(WILDCARD)
```

As a quick test I replaced the wildcard with an `EmptyTree` and, behold, the previous compilation errors in the type checker and namer go away. Could it be so simple? I assume the wildcard placeholder was there for a reason. I'll run some tests and update the PR if they are successful.